src/proto/device_path/text.rs

```diff
@@ -0,0 +1,122 @@
+//! `DevicePathToText` and `DevicePathFromText` Protocol
+
+use crate::{
+    proto::{device_path::DevicePath, Protocol},
+    unsafe_guid, CStr16, Char16,
+};
+
+/// Device Path to Text protocol.
+///
+/// This protocol provides common utility functions for converting device
+/// nodes and device paths to a text representation.
+#[repr(C)]
+#[unsafe_guid("8b843e20-8132-4852-90cc-551a4e4a7f1c")]
+#[derive(Protocol)]
+pub struct DevicePathToText {
+    convert_device_node_to_text: unsafe extern "efiapi" fn(
+        device_node: *const DevicePath,
+        display_only: bool,
+        allow_shortcuts: bool,
+    ) -> *const Char16,
+    convert_device_path_to_text: unsafe extern "efiapi" fn(
+        device_path: *const DevicePath,
+        display_only: bool,
+        allow_shortcuts: bool,
+    ) -> *const Char16,
+}
+
+impl DevicePathToText {
+    /// Convert a device node to its text representation.
+    ///
+    /// The `display_only` parameter controls whether the longer (parseable) or
+    /// shorter (display-only) form of the conversion is used. If `display_only`
+    /// is TRUE, then the shorter text representation of the display node is
+    /// used, where applicable. If `display_only` is FALSE, then the longer text
+    /// representation of the display node is used.
+    ///
+    /// The `allow_shortcuts` is FALSE, then the shortcut forms of text
+    /// representation for a device node cannot be used. A shortcut form
+    /// is one which uses information other than the type or subtype. If
+    /// `allow_shortcuts is TRUE, then the shortcut forms of text
+    /// representation for a device node can be used, where applicable.
+    ///
+    /// Returns `None` if `device_node` was NULL or there was insufficient memory.
+    pub fn convert_device_node_to_text(
+        &self,
+        device_node: &DevicePath,
+        display_only: bool,
+        allow_shortcuts: bool,
```

Comment:
To make call sites clearer, let's add some boolean wrapper types list this:

```rust
/// ...
pub struct DisplayOnly(pub bool);

/// ...
pub struct AllowShortcuts(pub bool);
```

Then you can also move the info about what `display_only` and `allow_shortcuts` do to the docstrings for those wrapper types so you don't have to repeat them in the `convert_device_node_to_text` and `convert_device_path_to_text` docstrings.

Reply:
It's done!😃